pl_examples/basic_examples/autoencoder.py

```diff
@@ -32,8 +32,6 @@
     from torchvision import transforms
 if _TORCHVISION_MNIST_AVAILABLE:
     from torchvision.datasets import MNIST
-else:
-    from tests.helpers.datasets import MNIST
```

Comment:
so it will fail if MNIST cannot be downloaded?
cc: @awaelchli

Reply:
So I tried removing this because we are in a deadlock

On one hand, we want to exclude the tests directory from the distributed packages
But on the other hand, we need the test MNIST helpers to run some of the examples

So we have 4 options:

1. Do not exclude tests from the distribution
2. Do not use our own MNIST implementation (this is what I explored with the current changes)
3. Move the MNIST implementation to the `pytorch_lightning` folder
4. Copy the MNIST implementation in the `example` file that uses it.

@Borda how do you think we should proceed?

Reply:
I think the 3 or 4 are fine, but I would prefer 4.

Reply:
I think the best is 4)

Reply:
I would be against 3). This is nothing related to PL itself.
I also think 4) is best.